test/e2e/manifest_test.go

```diff
@@ -39,10 +40,16 @@ var _ = Describe("Podman manifest", func() {
 	})
 
 	AfterEach(func() {
+		// if auth test fails, it will leave a registry running
+		if registry != nil {
+			_ = registry.Stop()
+		}
```

Comment:
If the registry is state related to a single test, I’d naively expect only that test to be dealing with that, without the callers having to care:

```go
		os.Setenv("PODMAN", podmanTest.PodmanBinary+" "+opts)
		registry, err = podmanRegistry.StartWithOptions(registryOptions)
		Expect(err).To(BeNil())
		defer func() {
			err := registry.Stop()
			Expect(err).To(BeNil())
		}()
```

Same for the environment variable. Is there some non-obvious reason why that won’t work, e.g. is there a risk of the test process crashing and not having a chance to run the `defer`? If so, it would be nice to add a comment. (Or is this a general practice of the e2e tests not to rely on `defer`, which everyone understands and I’m just unaware of the convention?)

Reply:
I see plenty of `defer()s` under `test/e2e`. The reason I didn't use it is that I'm not a seasoned Go programmer, and completely forgot about it. Thank you, I will submit a new PR (Monday) to address this.

Reply:
In that case, it’s just a small low-priority cleanup — the code as is works just fine.